### PR TITLE
[risk=no][RW-2474] Billing Project buffer error handling

### PR DESCRIPTION
**Pull request by ericsong**
"Error" Handling.

All its really doing is finding projects that have been stuck in the temporary states for too long and set them to ERROR.

The timeout for CREATING is 60m and ASSIGNING is 10m.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/db/model/BillingProjectBufferEntry.java [hunk 1]
```diff
@@ -84,7 +95,9 @@ public BillingProjectBufferStatus getStatusEnum() {
     return StorageEnums.billingProjectBufferStatusFromStorage(status);
   }
 
-  public void setStatusEnum(BillingProjectBufferStatus status) {
+  public void setStatusEnum(
+      BillingProjectBufferStatus status, Supplier<Timestamp> currentTimestamp) {
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/db/model/BillingProjectBufferEntry.java]: This is definitely on the fence for me. We don't really have business logic in these model classes anywhere else in the code AFAIK, but it also makes all the call sites pretty clean and ensures we don't miss any. Of course, someone could go straight to the `setStatus()` method and bypass this.

File: api/src/main/resources/workbench.yaml [hunk 2]
```diff
@@ -1078,6 +1078,22 @@ paths:
           schema:
             $ref: '#/definitions/ErrorResponse'
 
+  /v1/cron/cleanBillingBuffer:
+    get:
+      security: []
+      tags:
+        - billing
+        - cron
+      description: ind BillingProjectBufferEntries that have failed during the creation or assignment step and set their statuses to ERROR
```
Comment (calbach) [inline api/src/main/resources/workbench.yaml]: Find

File: api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java [hunk 3]
```diff
@@ -101,11 +110,34 @@ public void syncBillingProjectStatus() {
     }
   }
 
+  public void cleanBillingBuffer() {
+    Iterables.concat(
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java]: Nice

File: api/src/main/java/org/pmiops/workbench/db/model/BillingProjectBufferEntry.java (hunk 1 above)
Comment (ericsong) [reply api/src/main/java/org/pmiops/workbench/db/model/BillingProjectBufferEntry.java]: Yeah... I wasn't sure about this either but I decided the protection against misuse is worth it. `setStatus()` and `setLastStatusChangedTime()` are actually both private so the only way to update either is through this method.

**Pull request merged**